### ISSUE TITLE
global evaluations for querySelectorAll

### ISSUE THREAD

**Issue by jasonLaster** [RecordReplay/devtools#2679]
**Description:**

@andarist noticed while working on livechat that there are some inconsistencies with doucment.querySelectorAll and whether we are paused at a frame
https://replay.io/view?id=02d1b4aa-16f4-4cba-a00d-2863f5c8b70a&point=0&time=0&hasFrames=false

Here is a replay with 5 frame evaluations
https://replay.io/view?id=078b8882-c364-4352-b5b5-323fd4157a0e

**Comment by bhackett1024** [comment]
I don't understand from the second replay whether any of these evaluations are behaving incorrectly.  Doing an evaluation will behave differently depending on where in the recording we are paused and whether we are paused at a location inside an iframe's code (in which case the "document" will get the iframe's document).

The original problem which @Andarist ran into is in the replay below.  It looks like the debugger is doing the evaluation while we aren't paused at a point in the iframe's scripts, which would cause the querySelectorAll to return an empty list because we will do the evaluation against the outer window.  We should be able to use this replay to figure out how the debugger got into that state, but the lack of mouse events (https://github.com/RecordReplay/devtools/issues/2680) and rendering problems when replaying (https://github.com/RecordReplay/gecko-dev/issues/359) make it hard to see what's going on.

https://replay.io/view?id=3c28b86a-e92f-4ec7-8af8-8e5d04f62dfa

**Comment by jasonLaster** [comment]
@bhackett1024 good find. I didnt realize we had an iframe here.

**Comment by hbenl** [comment]
If you look at https://replay.io/view?id=3c28b86a-e92f-4ec7-8af8-8e5d04f62dfa carefully, you'll find that the timeline makes 3 jumps shortly before the `querySelectorAll` evaluation. It seems that these jumps are not triggered by mouse events, I'm pretty sure that they're caused by #2661 (there are global shortcuts for jumping to the previous/next paint point using left/right cursor keys and those were also triggered when using the cursor keys in an input element like the replay console), you can also see the corresponding keyboard events in the recording. As a result, the evaluation happened at a different execution point than expected.
In other words, I think that @bhackett1024 's theory is correct and the underlying cause was #2661 which was fixed in the meantime.
